Applications/Utils/MeshEdit/flipNodeCoordinates.cpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ *
+ */
+
+#include <memory>
+#include <string>
+#include <valarray>
+
+#include <tclap/CmdLine.h>
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+#include "BaseLib/BuildInfo.h"
+
+#include "MeshLib/Node.h"
+#include "MeshLib/Mesh.h"
+
+#include "MeshLib/IO/readMeshFromFile.h"
+#include "MeshLib/IO/writeMeshToFile.h"
+
+void flipNodeCoordinates(MeshLib::Mesh &mesh, std::array<int, 3> const& new_coord_ids)
+{
+    double new_xyz[3] = {};
+    for (MeshLib::Node* node : mesh.getNodes())
+    {
+        for (int i=0; i<3; i++)
+            new_xyz[i] = (*node)[new_coord_ids[i]];
+        node->updateCoordinates(new_xyz[0], new_xyz[1], new_xyz[2]);
+    }
+}
+
+static bool parseOrderString(std::string const& str_order, std::array<int, 3> &new_coord_ids)
+{
+    if (str_order.length()!=3)
+    {
+        ERR("Invalid argument for for the new order. The argument should contain three characters.");
+        return false;
+    }
+
+    for (std::size_t i=0; i<new_coord_ids.size(); i++)
+        new_coord_ids[i] = -1;
+
+    for (int i=0; i<3; i++)
+    {
+        if (str_order[i] == 'x')
+            new_coord_ids[i] = 0;
+        else if (str_order[i] == 'y')
+            new_coord_ids[i] = 1;
+        else if (str_order[i] == 'z')
+            new_coord_ids[i] = 2;
+        else {
+            ERR("Invalid argument for for the new order. The  given argument contains a character other than x, y, z.");
+            return false;
+        }
+    }
+
+    bool marked[3] = {false};
+    for (std::size_t i=0; i<new_coord_ids.size(); i++)
+    {
+        if (marked[new_coord_ids[i]])
+        {
+            ERR("Invalid argument for for the new order. The argument contains some character used more than once.");
+            return false;
+        }
+        marked[new_coord_ids[i]] = true;
+    }
+
+    return true;
+}
+
+int main(int argc, char *argv[])
+{
+    ApplicationsLib::LogogSetup logog_setup;
+
+    TCLAP::CmdLine cmd("Flips node coordinates", ' ', BaseLib::BuildInfo::git_describe);
+    TCLAP::ValueArg<std::string> input_arg("i", "input-mesh-file","input mesh file",true,"","string");
+    cmd.add( input_arg );
+    TCLAP::ValueArg<std::string> output_arg("o", "output-mesh-file","output mesh file",true,"","string");
+    cmd.add( output_arg );
+    TCLAP::ValueArg<std::string> new_order_arg("n", "new-coords-order", "the new order of coordinates (e.g. xzy for flipping XY plane to XZ plane)", true, "", "string");
```

Comment:
- yes. the given order is used to exchange coordinate values in the new mesh.
- probably "flipping" is wrong. I didn't have a better word for it. "Rotation" cannot be used because this tool will not support rotation with arbitrary angles. how about "swap" or "exchange"?